server/metadata/tenant.go

```diff
@@ -166,6 +166,24 @@ func (m *TenantManager) CreateOrGetTenant(ctx context.Context, txMgr *transactio
 	return m.createOrGetTenantInternal(ctx, tx, namespace)
 }
 
+// CreateTenant is a thread safe implementation of creating a new tenant. It returns the error if it already exists.
+func (m *TenantManager) CreateTenant(ctx context.Context, tx transaction.Tx, namespace Namespace) (err error) {
+	m.Lock()
+	defer m.Unlock()
+	namespaces, err := m.encoder.GetNamespaces(ctx, tx)
+
+	if id, found := namespaces[namespace.Name()]; found {
+		return api.Errorf(api.Code_CONFLICT, "namespace with same name already exists with id '%s'", fmt.Sprint(id))
+	}
+	for name, id := range namespaces {
+		if id == namespace.Id() {
+			return api.Errorf(api.Code_CONFLICT, "namespace with same id already exists with name '%s'", name)
+		}
+	}
+	_, err = m.createOrGetTenantInternal(ctx, tx, namespace)
```

Comment:
as you are not returning the tenant don't call this as this method is also creating a tenant object which is used to cache it. Replace this by executing the version bump and encoding the tenant information like this,
```
log.Debug().Str("tenant", namespace.Name()).Msg("tenant not found, creating")

// bump the version first
if err := m.versionH.Increment(ctx, tx); ulog.E(err) {
	return nil, err
}

if err := m.encoder.ReserveNamespace(ctx, tx, namespace.Name(), namespace.Id()); ulog.E(err) {
	return nil, err
}
```

Reply:
updated.